packages/grid/test/column-min-width.common.ts

```diff
@@ -0,0 +1,324 @@
+import { expect } from '@esm-bundle/chai';
+import { fixtureSync, nextRender, oneEvent } from '@vaadin/testing-helpers';
+import sinon from 'sinon';
+import type { Grid } from '../src/vaadin-grid.js';
+import type { GridColumn } from '../src/vaadin-grid-column.js';
+import type { GridColumnGroup } from '../src/vaadin-grid-column-group.js';
+import { fire, flushGrid, getRowCells, getRows, infiniteDataProvider } from './helpers.js';
+
+interface ExpectedWidths {
+  pattern: RegExp;
+  values: number[];
+}
+function expectColumnWidthsToBeOk(
+  columns: NodeListOf<GridColumn<any> | GridColumnGroup<any>>,
+  expectedWidths: ExpectedWidths[] = [
+    { pattern: /max\((\d+)px, (\d+)px\)/u, values: [50, 71] },
+    { pattern: /(\d+)[px]?/u, values: [114] },
+    { pattern: /(\d+)[px]?/u, values: [84] },
+    { pattern: /(\d+)[px]?/u, values: [107] },
+  ],
+  delta = 5,
+) {
+  // Allowed margin of measurement to keep the test from failing if there are small differences in rendered text
+  // width on different platforms or if there are small changes to styles which affect horizontal margin/padding.
+  expectedWidths.forEach((expectedWidth, index) => {
+    const colWidth = columns[index].width;
+    expect(colWidth).to.be.not.undefined;
+    expect(colWidth).to.be.not.null;
+    const split = colWidth!.split(expectedWidth.pattern);
+    for (let indexValue = 0; indexValue < expectedWidth.values.length; indexValue++) {
+      const widthValue: string = split[indexValue + 1];
+      const expectedWidthValue = expectedWidth.values[indexValue];
+      const columnWidth = parseInt(widthValue);
+      expect(columnWidth).to.be.closeTo(expectedWidthValue, delta);
+    }
+  });
+}
+
+describe('column auto-width', () => {
```

Comment:
Is there a reason to have all min-width -specific tests in this separate test file instead of including, for example, the auto-width test to `column-auto-width.common.js` (under "min-width" suite where necessary)?

As an example, adding the following (adjusted) test from this PR to `column-resizing.common.js` right under "should resize the last non-hidden child column" would save us from replicating the same test setup here:

```js
it('should resize the child columns with min-width', () => {
  const columns = grid.querySelectorAll('vaadin-grid-column');
  columns[0].minWidth = '120px';
  columns[1].minWidth = '70px';

  const headerRows = getRows(grid.$.header);
  const groupCell = getRowCells(headerRows[0])[0];
  const handle = groupCell.querySelector('[part~="resize-handle"]');

  const cell = getRowCells(headerRows[1])[1];
  const rect = cell.getBoundingClientRect();
  const options = { node: handle };
  expect(cell.clientWidth).to.equal(100);
  fire('track', { state: 'start' }, options);
  fire('track', { state: 'track', x: rect.right + (direction === 'rtl' ? -50 : 50), y: 0 }, options);

  expect(cell.clientWidth).to.equal(direction === 'rtl' ? 70 : 150);
  expect(groupCell.clientWidth).to.equal(direction === 'rtl' ? 190 : 270);
});
```

Reply:
For the resizing, I think there is no particular reason. For the other tests, they are using `expectColumnWidthsToBeOk` with a regexp.
`auto-width` is not what I'm testing that's the reason it's not in the `auto-width` test file.
For example, I need to add at least a test case without `auto-width`(that could be in an other file or in `min-width`).

Let me add all the needed tests then we can check if they need to be in the existing test files or in one specific file.